lib/bolt/config/transport.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+require 'bolt/util'
+
+module Bolt
+  class Config
+    class Transport
+      attr_reader :config
+
+      def initialize(data = {}, boltdir = nil)
+        assert_hash(data)
+        @config = Bolt::Util.deep_merge(defaults, filter(data))
+        @boltdir = boltdir
+        validate
+      end
+
+      def [](key)
+        @config[key]
+      end
+
+      def to_h
+        @config
+      end
+
+      def fetch(*args)
+        @config.fetch(*args)
+      end
+
+      def include?(args)
+        @config.include?(args)
+      end
+
+      def config=(data)
+        assert_hash(data)
+        @config = data
+        validate
+      end
+
+      def merge(*data)
```

Comment:
This method currently only accepts hashes when merging transport config. I thought it might be nice to have it also accept `self.class`, but the only place merging happens is when merging the group cache config and target config, which are both hashes.